### PR TITLE
add tests for `on duplicate key update` for keyless tables

### PR DESCRIPTION
**Pull request by JCOR11599**
small change to in memory table edit accumulators to only remove rows once from `adds` and `deletes` for `Insert()` and `Delete()`

companion pr: https://github.com/dolthub/dolt/pull/5514

### CODE REVIEW

File: enginetest/queries/insert_queries.go [hunk 1]
```diff
@@ -1606,6 +1606,76 @@ var InsertScripts = []ScriptTest{
...
+			{
+				Query: `select i, j from t order by i`,
+				Expected: []sql.Row{
+					{0, "third"},
+				},
+			},
+		},
+	},
+	{
+		Name: "insert on duplicate key for keyless table multiple unique columns",
```
Comment (max-hoffman) [inline enginetest/queries/insert_queries.go:1636]: Only additional tests I can think of is maybe something like `insert into t values (0,0,0), (0,0,0),(0,0,1),(0,0,1) on duplicate key update c3 = 1`. And then a variation with nulls maybe? `insert into t values (0,null,0), (0,null,0) on duplicate key update c2 = null/0`

Comment (max-hoffman) [reply enginetest/queries/insert_queries.go:1636]: sometimes ordinals cause problems, maybe one with `unique key (c3,c2,c1)` could trigger some edge case

**Review by max-hoffman — APPROVED** [review]
lgmt nice fixes on both the dolt and GMS sides. Did not dig in to understand the early return from the memory TEA, also a couple test suggestions

File: enginetest/queries/insert_queries.go (hunk 1 above)
Comment (JCOR11599) [reply enginetest/queries/insert_queries.go:1636]: The tests with nulls and ordinals works, so I added those.
The batched inserts do not work in GMS, but work fine in dolt, so I added a skipped test for that.
I have a fix for that, but it breaks other stuff, so I think a follow up PR and a new tracking would be better.
